app/models/user.rb

```diff
@@ -75,6 +76,17 @@ def self.reset_token_expired?(sent_at)
     Time.current > (sent_at.in(RESET_TOKEN_VALIDITY_PERIOD))
   end
 
+  def generate_session_token!(extended_session: false)
+    digest = User.generate_digest(SecureRandom.alphanumeric(40))
```

Comment:
Since we'll need only the digest we can leverage a performance boost by using `random_bytes` or `gen_random`?

Reply:
FYI here's a benchmark, for a `500_000` trials for each case on my local setup `gen_random` is approximetly 5.7 times faster:
![image](https://user-images.githubusercontent.com/29759616/191486742-aebae914-a7a6-4f45-adc7-abc7153090d0.png)